app/routes.py

```diff
@@ -1,2 +1,53 @@
+from app import db
+from app.models.planet import Planet
 from flask import Blueprint
+from flask import request, make_response, jsonify
+
+
+planet_bp = Blueprint("planet", __name__, url_prefix="/planets")
+
+
+@planet_bp.route("/", methods=["GET", "POST"])
+def handle_all_planets():
+    if request.method == "GET":
+        name_query = request.args.get("name")
+        if name_query:
+            planets_result = Planet.query.filter_by(name=name_query)
+        else:
+            planets_result = Planet.query.all()
+        return jsonify([p.planet_dict() for p in planets_result]), 200
+    elif request.method == "POST":
+        request_body = request.get_json()
```

Comment:
Some validation to make sure the `request_body` is valid (has required fields) would be good.